### ISSUE TITLE
I have some question about the code and the corresponding paper descreption.

### ISSUE THREAD

**Issue by db12138** [shmsw25/qa-hard-em#12]
1. first question:Why a random condition is set when you choose a loss , which should always be the max nagative log(Probability)  of all "solutions" descrebed in the paper (I mean multi-mention setting ).
![image](https://user-images.githubusercontent.com/23103161/79409979-ca785e80-7fd1-11ea-8c0a-2d162cad32ee.png)
2.second question: training data and predict data go through the same model.forward function. and there no "solutions" when doing predict . and When doing training, the loss should be chosen accroding to one of the "solutions" with max nagetive log(probability) . But I cant find  which variable refering to "solutions" in the code.  could you figure that out for me? 
Looking forward to your reply. thanks.

**Comment by shmsw25** [comment]
Hi @db12138, thanks for your interest.

1. We perform annealing that uses MML in the beginning of the training and gradually use a hard EM as training proceeds. The details are described in the last paragraph of Section 5.1. in the paper. 

2. In multi-mention reading comprehension or open-domain QA, a *solution* is defined as a *span* from the passage. You can see the model uses a start and end position of the span in the passage [here](https://github.com/shmsw25/qa-hard-em/blob/master/modeling.py#L410).
